### PR TITLE
Tao 871 section category

### PR DESCRIPTION
**Pull request by ssipasseuth**
requires https://github.com/oat-sa/tao-core/pull/543

This PR enable adding a category on the qti test section from the test authoring tool. The categories are propagated and saved to child itemRefs.

To validate the AC on the parent user story https://oat-sa.atlassian.net/browse/TAO-871
1.  set exitButton = true in testRunner.conf.php
2.  create a test and add the special category "x-tao-option-exit"
3. compile and execute the delivery -> the section having that exit code should have the exit button displayed, the section that does not have it should have no button "up there"

### CODE REVIEW

**Comment by ssipasseuth** [comment]
Made a last improvement, ready for review now.

File: views/js/controller/creator/helpers/sectionCategory.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            //the categories that are only partially covered on the section level : complementary of "propagated"
+            var partial = _.difference(union, propagated);
+            
+            return {
+                all : union.sort(),
+                propagated : propagated.sort(),
+                partial : partial.sort()
+            };
+        }else{
+            errorHandler.throw(_ns, 'invalid tool config format');
```
Comment (krampstudio) [inline views/js/controller/creator/helpers/sectionCategory.js]: Where are the errors listened ?

File: views/js/controller/creator/views/section.js [hunk 2]
```diff
@@ -145,12 +151,19 @@ function($, _, uri, actions, itemRefView, rubricBlockView, templates, qtiTestHel
...
+                            defaultItemData = {};
+                            
                         if(model.itemSessionControl && !_.isUndefined(model.itemSessionControl.maxAttempts)){
+                            
                             //for a matter of consistency, the itemRef will "inherit" the itemSessionControl configuration from its parent section
                             defaultItemData.itemSessionControl = _.clone(model.itemSessionControl);
                         }
                         
+                        //the itemRef should also "inherit" the categories set at the item level
+                        var categories = sectionCategory.getCategories(model);
```
Comment (krampstudio) [inline views/js/controller/creator/views/section.js]: the `categories` variable is redeclared

File: views/js/controller/creator/helpers/sectionCategory.js (hunk 1 above)
Comment (ssipasseuth) [inline views/js/controller/creator/helpers/sectionCategory.js]: It is for logging purpose only and offer an opportunity to manage such exception later on.

File: views/js/controller/creator/views/section.js (hunk 2 above)
Comment (ssipasseuth) [inline views/js/controller/creator/views/section.js]: oops

**Pull request merged**